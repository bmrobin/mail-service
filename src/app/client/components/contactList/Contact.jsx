import React from 'react';
import * as propTypes from 'prop-types';
import { Button } from 'react-bootstrap';
import './contact.css';

export default class Contact extends React.Component {

  constructor(props) {
    super(props);
  }

  deleteHandler() {
    this.props.delete(this.props.contact);
  }

  render() {
    return (
      <tr>
        <td>{this.props.contact.emailAddress}</td>
        <td className="action-column">
          <Button bsSize="sm" className="delete-button" onClick={() => this.deleteHandler()}>X</Button>
        </td>
      </tr>
    );
  }
}

Contact.propTypes = {
  delete: propTypes.func,
  contact: propTypes.shape({
    emailAddress: propTypes.string.isRequired
  }).isRequired
};
