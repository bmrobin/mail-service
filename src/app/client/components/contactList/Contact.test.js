import React from 'react';
import { shallow } from 'enzyme';
import { Button } from 'react-bootstrap';
import Contact from './Contact';

describe("Contact Component", () => {

  test("should render component", () => {
    const contact = {
      emailAddress: 'bmrobin@mail.com'
    };
    const component = shallow(<Contact contact={contact} />);
    expect(component.debug()).toMatchSnapshot();
  });

  test("should invoke delete callback when clicked", () => {
    const contact = {
      emailAddress: 'bmrobin@mail.com'
    };
    const mockDeleteFn = jest.fn();
    const component = shallow(<Contact contact={contact} delete={mockDeleteFn} />);
    component.find(Button).simulate('click');
    expect(mockDeleteFn).toHaveBeenCalledWith(contact);
  });

});
