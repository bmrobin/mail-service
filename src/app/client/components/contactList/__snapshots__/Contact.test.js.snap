// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Component should render component 1`] = `
"<tr>
  <td>
    bmrobin@mail.com
  </td>
  <td className=\\"action-column\\">
    <Button bsSize=\\"sm\\" className=\\"delete-button\\" onClick={[Function: onClick]} active={false} block={false} disabled={false} bsStyle=\\"default\\" bsClass=\\"btn\\">
      X
    </Button>
  </td>
</tr>"
`;
